### PR TITLE
docs: Add project origin summary

### LINKED ISSUE #449: Add a "why python?" section to the documentation
This needs to come from @matthewfeickert 's GitLab MR on the super-secret PUB note we're writing.

### PR DESCRIPTION
**Pull request by matthewfeickert**
# Description

Resolves #449

Add some historical context for why Python was chosen for the project language and how the project and dev team formed. This is mostly so we don't forget (our roots :slightly_smiling_face: ).

Finally making this PR has been on the back burner ever since I got interviewed for this [IRIS-HEP article](https://iris-hep.org/2020/09/17/fellows.html) and to make it easier for the reporter I compiled this list of links:

> Follow up RE: Matthew’s DIANA/HEP Fellowship:
> * (May, 2017) [Fellowship announcement tweet](https://twitter.com/SMUPhysics/status/861584474638766080)
> * (January, 2018) [First tweet mentioning the idea of pyhf](https://twitter.com/lukasheinrich_/status/956809112674885632) and [connection to my fellowship](https://twitter.com/lukasheinrich_/status/956809114696417280)
> * (October, 2018) [Last tweet to talk about pyhf as a DIANA/HEP related project](https://twitter.com/KyleCranmer/status/1052280418278498304) (after that became detached, then funded by IRIS-HEP and part of Scikit-HEP)
> * (April, 2018) [DIANA/HEP Fellowship Summary Report](https://github.com/matthewfeickert/DIANA-Proposal-Feickert/blob/dac5181e7e87e747fbdb3c5a6d0201f723fa4670/summary_report/summary_report.pdf)
> * (October, 2018) [DIANA/HEP Topical meeting talk on Fellowship](https://matthewfeickert.github.io/talk-DIANA-HEP-pyhf-2018/index.html?p=talk.md#1) (just ended up being a pyhf talk at that point)

@lukasheinrich @kratsg I'll need your help to make sure this doesn't become revisionist history though. :P

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Add 'Why use Python?' and 'How did pyhf get started?` questions
   - Add historical context and give motivations for the project
```

### CODE REVIEW

**Comment by matthewfeickert** [comment]
Relevant RTD build: https://pyhf--1415.org.readthedocs.build/en/1415/faq.html#why-use-python

File: docs/faq.rst [hunk 1]
```diff
@@ -78,6 +80,52 @@ If the journal you are publishing in requires you to use ``textsc`` for software
...
+frameworks allowed them to be quite effective tools for statistical fits.
+As the frameworks considered in this study (TensorFlow, PyTorch, MXNet) all provided
+low-level Python APIs to the libraries this made Python an obvious choice for a common
+high-level control language.
+Given all these considerations, Python was chosen as the development language.
+
+How did ``pyhf`` get started?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In 2017 Lukas Heinrich was discussing with colleauge Andy Buckley how it would be convienent to
```
Comment (lukasheinrich) [inline docs/faq.rst:103]: Andy Buckley -> Holger Schulz.  Maybe add that this resulted in paper w/ Jessica and Ye-Ling

Comment (matthewfeickert) [reply docs/faq.rst]: Revised. :+1: If you think the paper mention should get moved from where I put it for now,  can you make comments in the PR to indicate where?

Comment (matthewfeickert) [reply docs/faq.rst]: (I don't know the full history of this paper like you do, so this paper might have already been in progress when pyhf first got started and needs to get bumped up so it doesn't sound like a result _of_ pyhf).

**Review by kratsg — APPROVED** [review]

**Comment by kratsg** [comment]
There was probably some point in early 2018 (I'm thinking February when I was at CERN w/ Lukas).

**Comment by matthewfeickert** [comment]
> There was probably some point in early 2018 (I'm thinking February when I was at CERN w/ Lukas).

Sorry, there was probably some point in early 2018 that what happened?

**Comment by kratsg** [comment]
> Sorry, there was probably some point in early 2018 that what happened?

well Lukas and I had a late night at CERN revolving around pyhf + gpsys related stuff. It was either around Nov 2017 or Jan-Feb 2018 before my thesis writing period in March-April.

**Comment by matthewfeickert** [comment]
> It was either around Nov 2017 or Jan-Feb 2018 before my thesis writing period in March-April.

Hm, from the timeline I've reconstructed from emails and git commit timestamps it seems that it probably would have had to be February from the earliest. Do you want to make edits that add in when you think things happened, as I wasn't involved in these discussions?

**Comment by kratsg** [comment]
> Hm, from the timeline I've reconstructed from emails and git commit timestamps it seems that it probably would have had to be February from the earliest. Do you want to make edits that add in when you think things happened, as I wasn't involved in these discussions?

I can't remember too much... but I know we ended up in a Table in R1 for a while late at night just working on something related to MBJ + pyhf at the time, and that's probably what was I guess my ramp-up meeting to pyhf then.

File: docs/faq.rst [hunk 2]
```diff
@@ -78,6 +80,54 @@ If the journal you are publishing in requires you to use ``textsc`` for software
...
+in Python using machine learning libraries to drive the computation.
+Continuing the discussion in Lukas's office, Lukas showed Matthew that the core statistical
+machinery could be implimented rather succinctly, and that night
+`proceeded to do so <https://github.com/scikit-hep/pyhf/commit/fd32503fb760f070a4047cb867757458b1687599>`_
+dubbing the project ``pyhf``.
+
+.. tweet:: https://twitter.com/lukasheinrich_/status/956809112674885632
+
+Matthew joined him on the project to begin development and by April 2018 Giordon Stark had
+learned about the project and began making contributions, quickly becoming
```
Comment (matthewfeickert) [inline docs/faq.rst:123]: > I can't remember too much... but I know we ended up in a Table in R1 for a while late at night just working on something related to MBJ + pyhf at the time, and that's probably what was I guess my ramp-up meeting to pyhf then.

Okay. I had this bit in about April as the first PR from you was PR #104 on 2018-04-10, but @kratsg please feel free to edit any and all of this to fill things in more. Additionally, if @lukasheinrich had more memories then those should go in as well. :+1:

**Comment by matthewfeickert** [comment]
[`sphinxcontrib.twitter`](https://pypi.org/project/sphinxcontrib.twitter/) is now unsupported and doesn't support Sphinx `v4.x+` so drop it and instead of embedding @lukasheinrich's [original tweet](https://twitter.com/lukasheinrich_/status/956809112674885632) 

[original_tweet](https://twitter.com/lukasheinrich_/status/956809112674885632)

just link to it instead.

**Comment by matthewfeickert** [comment]
Relevant page for review: https://pyhf--1415.org.readthedocs.build/en/1415/faq.html#why-use-python

**Comment by matthewfeickert** [comment]
I'm going to merge this in for now, but I'm happy to revise this as much as needed by either of you. :+1:

**Pull request merged**